### ISSUE TITLE
Manage `@EntityGraph`s in Panache `PanacheQuery`

### ISSUE THREAD

**Issue by luca-bassoricci** [quarkusio/quarkus#29852]
### Description

`PanacheQuery` expose `Hibernate` session filters but there isn't supports for `@EntityGraph`s.
Evaluate to expose `@EntityGraph` in `PanacheQuery`.

### Implementation ideas

A trivial implementation may add some methods to `PanacheQuery` as `#fetchGraph(String graphName)`/`#loadGraph(String graphName)` or `#graph(String graphType, String graphName)`.
IDK If more complex actions are required during deployment (for example recording `@NamedEntityGraph`s to be setted in `io.quarkus.hibernate.orm.panache.common.runtime.PanacheHibernateRecorder`)

**Comment by Plawn** [comment]
Maybe this could be done at the same time :)  #25333

**Comment by omasseau** [comment]
As ~~@EntityGraph~~ @NamedEntityGraph is not supported what is the alternative to eager load an associated entity for a specific query with Panache ? I can't find this information in the documentation.

**Comment by FroMage** [comment]
I could not find any annotation named `@EntityGraph`, perhaps you meant `@NamedEntityGraph`?

We could expose this in `PanacheQuery` for Panache 1, but I'd rather support something like being able to apply a named entity graph to a query method for Panache 2:

```java
@Entity
@NamedEntityGraph(name = "MyGraph", … )
public class MyEntity {
 @Repository
 public interface Repo {
  @Find
  @FetchGraph(MyEntity_.MyGraph)
  public List<MyEntity> findAll();
 }
}
```

Do you already have something like that in Jakarta Data @gavinking ? I could not find it in the spec.

**Comment by gavinking** [comment]
There is something coming in JPA4.

**Comment by FroMage** [comment]
Oh, OK. So we only have to do something for Panache 1, if we want. Thanks.

**Comment by omasseau** [comment]
> I could not find any annotation named `@EntityGraph`, perhaps you meant `@NamedEntityGraph`?

Yes sorry I meant [@NamedEntityGraph](https://thorben-janssen.com/jpa-21-entity-graph-part-1-named-entity/)  indeed ;)
